### PR TITLE
KNOWAGE-8144 Update

### PR DESCRIPTION
**Pull request by dbulatovicx32**
No longer catching an exception as requested.
Informing the user in case file does not exist.

### CODE REVIEW

File: knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java [hunk 1]
```diff
@@ -77,19 +77,20 @@ public void doService() {
...
-					HttpServletResponse response = getHttpResponse();
-					response.setHeader("Content-Disposition", "attachment; filename=\"" + file.getName() + "\";");
-					byte[] content = SpagoBIUtilities.getByteArrayFromInputStream(fis);
-					response.setContentLength(content.length);
-					response.getOutputStream().write(content);
-					response.getOutputStream().flush();
-				} catch (Throwable t) {
-					logger.error("Error getting file", t);
+				if (!file.exists()) {
 					writeBackToClient(404, "Error getting file with name, not found", false, null, "text/plain");
```
Comment (davide-zerbetto) [inline knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java:83]: put "File not found" as message

File: knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java [hunk 2]
```diff
@@ -77,19 +77,20 @@ public void doService() {
...
 				}
 
+				FileInputStream fis = new FileInputStream(file);
+				HttpServletResponse response = getHttpResponse();
+				response.setHeader("Content-Disposition", "attachment; filename=\"" + file.getName() + "\";");
+				byte[] content = SpagoBIUtilities.getByteArrayFromInputStream(fis);
+				response.setContentLength(content.length);
+				response.getOutputStream().write(content);
+				response.getOutputStream().flush();
+
```
Comment (davide-zerbetto) [inline knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java:93]: this part must be inserted within an else block:
if (!file.exists()) {
    writeBackToClient...
}{
   return file content
}

**Review by davide-zerbetto — CHANGES_REQUESTED** [review]

File: knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java (hunk 1 above)
Comment (dbulatovicx32) [reply knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java]: Updated message.

File: knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java [hunk 3]
```diff
@@ -77,17 +77,18 @@ public void doService() {
 			} else if (OPER_DOWNLOAD.equalsIgnoreCase(operation)) {
 				freezeHttpResponse();
 				String fileName = (String) getAttribute("fileName");
+
 				File file = getFile(fileName);
-				try (FileInputStream fis = new FileInputStream(file)) {
+				if (!file.exists()) {
+					writeBackToClient(404, "File not found.", false, null, "text/plain");
+				} else {
+					FileInputStream fis = new FileInputStream(file);
```
Comment (davide-zerbetto) [inline knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java:85]: You removed the try-with-resources block, stream will stay open intead of being closed (every stream must be closed).
Use:
try (FileInputStream fis = new FileInputStream(file)) {
  HttpServletResponse response = getHttpResponse();
  response.setHeader("Content-Disposition", "attachment; filename=\"" + file.getName() + "\";");
  byte[] content = SpagoBIUtilities.getByteArrayFromInputStream(fis);
  response.setContentLength(content.length);
  response.getOutputStream().write(content);
  response.getOutputStream().flush();
}

**Review by davide-zerbetto — CHANGES_REQUESTED** [review]

File: knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java (hunk 3 above)
Comment (dbulatovicx32) [reply knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java]: Wrapped the code with try block.

**Review by davide-zerbetto — APPROVED** [review]

**Pull request merged**